platform/openide.util.ui/src/org/openide/util/ImageUtilities.java

```diff
@@ -1112,6 +1201,7 @@ public ToolTipImage(Icon delegateIcon, String toolTipText, URL url, int imageTyp
             this.delegateIcon = delegateIcon;
             this.toolTipText = toolTipText;
             this.url = url;
+            this.imageId = null;
```

Comment:
Do we not need to pass the imageId along here?

Reply:
The other constructor is called from `assignToolTipToImageInternal` that reads `imageID` property from the image. That one is called from the `image2Icon`, which is used by `mergeImages`.In addition 
```
icon2Image(image2Icon(image_with_imageId))
```
ought to provide an image that still returns `imageID` property. So I think yes, TooltipImage should remember it.